feedhenry/fh-ios-sdk
# Convert to modern Objective-C syntax

**Motivation:**
The sdk used old syntax and constructs which were superseded by modern objective-c syntax. 
After applying Xcode magic to auto-convert to use  'modern collection literas/instancetype/NS_ENUM' have made the following changes:
- refactor to maximize property acccess(if visible on the outside) and proper overload in getters where dynamic properties are computed.
- added @protected access for those instance vars that accessed to the sublclasses so it's explicit.
- remove ```get`` prefixes from methods, naming not much used in obj-c
- unneeded @synthesize constructs
- use the common GCD approach for singleton creation

>Note: most of the changes where internal API changes not visible to the outside world

**To test:**
- Simple open the fh-ios-sdk.xcodeproj and run Product->Tests. The tests should build and run successfully.
- Invoke the build.sh script and once the fwk is build, open each demo in the /demos directory. They should work as expected (Note: that you need to adjust fhconfig.plist with your cloud FH app's id etc).

@wei-lee mind to review?
